tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/HotChocolateCommon.cs

```diff
@@ -45,6 +45,65 @@ internal static Scope CreateScopeFromExecuteAsync<T>(Tracer tracer, in T request
             return scope;
         }
 
+        internal static void UpdateScopeFromExecuteAsyncV13<TOperationContext>(Tracer tracer, in TOperationContext context)
+        where TOperationContext : IOperationContextV13
+        {
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return;
+            }
+
+            var scope = tracer.InternalActiveScope;
+            var span = scope?.Span;
+            if (span == null || span.OperationName != ExecuteOperationName)
+            {
+                // not in a Hotchocolate execution span
+                return;
+            }
+
+            try
+            {
+                var operation = context.Operation;
+                var operationType = operation.OperationType.ToString();
+                if (span.Tags is GraphQLTags tags)
+                {
+                    tags.OperationName = operation.Name;
+                    span.ResourceName = $"{operationType} {tags.OperationName ?? "operation"}";
+                    tags.OperationType = operationType;
+                }
+                else
+                {
+                    var operationName = span.GetTag(Trace.Tags.GraphQLOperationName);
+                    span.ResourceName = $"{operationType} {operationName ?? "operation"}";
+                    span.SetTag(Trace.Tags.GraphQLOperationType, operationType);
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error updating HotChocolate scope.");
+            }
+        }
+
+        private static string GetNameStringValue(object nameValue)
+        {
+            // Get the string value of the NameString
+            // The NameString value can be either another NameString or a string
+            while (nameValue != null)
```

Comment:
For this, that's weird because in the officiel source code the NameString can only have a string as a Value, but when looking at runtime, in the Value of the first NameString, that's another NameString, and for this one, it's a string value. So I would say a level of 2.

Reply:
Weird indeed. If we expect 2 levels maybe remove the while? I haven't found any case where it should be an issue but seeing the while raised a infinite loop alert to me :) even though I agree, for no reason